tests/PHPStan/Analyser/nsrt/mixed-subtract.php

```diff
@@ -53,4 +53,9 @@ function subtract(mixed $m, $moreThenFalsy) {
 		assertType('mixed', $m);
 		assertType('bool', (bool) $m); // could be true
 	}
+
+	if ($m != 0 && !is_array($m) && $m != null && !is_object($m)) { // subtract more types then falsy
```

Comment:
the test in the above if is still not able to narrow the type, but I think its related to that we compare 2 variables.
when comparing against literals we at least were able to trigger the bug in the previous `StaticTypeFactory::falsey()->equals($this->subtractedType)` logic